src/main/java/seedu/address/logic/commands/SetCommand.java

```diff
@@ -0,0 +1,149 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SLACK;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TELEGRAM;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.contact.Contact;
+import seedu.address.model.person.contact.ContactType;
+import seedu.address.ui.MainPanelName;
+
+/**
+ * Sets a Contact of a Person.
+ */
+public class SetCommand extends Command {
+    public static final String COMMAND_WORD = "set";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sets a person's Contacts. "
+            + "Parameters: "
+            + PREFIX_EMAIL + "EMAIL "
+            + PREFIX_PHONE + "PHONE "
+            + PREFIX_SLACK + "SLACK"
+            + PREFIX_TELEGRAM + "TELEGRAM"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_EMAIL + "e0123456@u.nus.edu "
+            + PREFIX_PHONE + "87654321"
+            + PREFIX_SLACK + "johndoe "
+            + PREFIX_TELEGRAM + "@johndoe";
+
+    public static final String MESSAGE_UPDATE_SUCCESS = "Contact updated.";
+
+    private final SetContactDescriptor setContactDescriptor;
+
+    /**
+     * Instantiates a SetCommand object.
+     *
+     * @param setContactDescriptor Container for the Contacts to be updated
+     */
+    public SetCommand(SetContactDescriptor setContactDescriptor) {
+        this.setContactDescriptor = setContactDescriptor;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        // get the selected person
+        Person toUpdate;
+        if (model.getSelectedPerson().get() != null) {
+            toUpdate = model.getSelectedPerson().get();
+        } else {
+            throw new CommandException("no person selected");
+        }
```

Comment:
The person to update is never `null` since this command is run in Detail panel. Should we use assertion here instead of exception? because if `toUpdate` is null, then something wrong with the programmer instead of users.

```suggestion
        Person selectedPerson = model.getSelectedPerson();
        assert selectedPerson != null;
```